index.js

```diff
@@ -2,30 +2,69 @@ import execa from 'execa';
 import pMemoize from 'p-memoize';
 import delay from 'delay';
 
+const powershell = (async (command, args) => {
+	if (typeof (command) !== 'string') {
+		throw new TypeError('Invalid command');
+	}
+
+	return execa(command, args, {shell: 'powershell'});
+});
+
 const getDevice = pMemoize(async () => {
-	if (process.platform !== 'darwin') {
-		throw new Error('macOS only');
+	if (process.platform === 'darwin') {
+		const {stdout} = await execa('networksetup', ['-listallhardwareports']);
+		const result = /Hardware Port: Wi-Fi\nDevice: (?<device>en\d)/.exec(stdout);
+		if (!result) {
+			throw new Error('Couldn\'t find a Wi-Fi device');
+		}
+
+		return result.groups.device;
 	}
 
-	const {stdout} = await execa('networksetup', ['-listallhardwareports']);
+	if (process.platform === 'win32') {
+		const {stdout} = await powershell('Get-NetAdapter', ['-Name Wi-Fi | Format-List -Property PnPDeviceID']);
+		// stdout: `PnPDeviceID : USB\VID_0BRA&PID_8387\00C0CA7B646E`
+		const result = /(?<=PnPDeviceID : ).*/.exec(stdout);
+
+		if (!result) {
+			throw new Error('Couldn\'t find a Wi-Fi device');
+		}
 
-	const result = /Hardware Port: Wi-Fi\nDevice: (?<device>en\d)/.exec(stdout);
-	if (!result) {
-		throw new Error('Couldn\'t find a Wi-Fi device');
+		return result[0];
 	}
 
-	return result.groups.device;
+	throw new Error('macOS or Windows only');
 });
 
 const isOn = async device => {
-	const {stdout} = await execa('ifconfig', [device]);
-	return stdout.includes('status: active');
+	if (process.platform === 'darwin') {
+		const {stdout} = await execa('ifconfig', [device]);
+		return stdout.includes('status: active');
+	}
+
+	if (process.platform === 'win32') {
+		const {stdout} = await powershell('Get-NetAdapterAdvancedProperty', ['-Name Wi-Fi -RegistryKeyword RFOff -AllProperties | Format-List -Property RegistryValue']);
+		// stdout: `RegistryValue : {0}`
+		const status = /(?<={).+?(?=})/.exec(stdout);
+		return !Number.parseInt(status, 10);
+	}
 };
 
 const toggleDevice = async (device, turnOn) => {
-	await execa('networksetup', ['-setairportpower', device, (turnOn ? 'on' : 'off')]);
+	if (process.platform === 'darwin') {
+		await execa('networksetup', [
+			'-setairportpower',
+			device,
+			turnOn ? 'on' : 'off'
+		]);
+	} else if (process.platform === 'win32') {
+		const setStatus = turnOn ? 0 : 1;
+		console.log('Testing...'); // Odd Issue: Without this log statement, tests fail on Windows 10
```

Comment:
You need to figure this one out. We cannot `console.log` in the published module.

Reply:
I've figured the issue by using the verbose mode on ava and it turned out to be `✖ Timed out while running tests` error. From that reason, I've added 30 second timeout to the test. It may not be the best solution though, do you think there is a better way to handle that?